subprojects/internal-testing/src/main/groovy/org/gradle/test/precondition/RequiresExtension.groovy

```diff
@@ -69,12 +69,19 @@ class RequiresExtension implements IAnnotationDrivenExtension<Requires> {
     /**
      * Common method for handling annotations.
      */
-    void visitAnnotation(Requires annotation, SpecElementInfo feature) {
-        def predicateClassNames = annotation.value() as List
+    void visitAnnotation(Requires annotation, SpecElementInfo specOfFeature) {
+        List<Class> predicateClassNames = annotation.value() as List<Class>
+
+        if (specOfFeature.parent instanceof SpecInfo) {
+            predicateClassNames += (specOfFeature.parent as SpecInfo).annotations
+                .findAll { it instanceof Requires }
```

Comment:
🤔 You could use `getAnnotation(Requires)` or `getAnnotationsByType(Requires)` instead, right?